pkg/reconciler/taskrun/validation.go

```diff
@@ -85,3 +105,217 @@ func extractPlatform(tr *tektonapi.TaskRun) (string, error) {
 	}
 	return "", errMissingPlatformParameter
 }
+
+// validateNumericValue validates a string represents a valid integer within the specified range
+// Validation rules:
+// - Must be a valid integer parseable by strconv.Atoi
+// - Must be between 1 and maxValue inclusive
+//
+// Parameters:
+// - value: The string value to validate
+// - maxValue: The maximum allowed value
+//
+// Returns:
+// - int: The validated integer value
+// - error: errInvalidNumericValue if value is invalid or out of range
+func validateNumericValue(value string, maxValue int) (int, error) {
+	num, err := strconv.Atoi(value)
+	if err != nil || num != max(0, min(num, maxValue)) {
+		return 0, errInvalidNumericValue
+	}
+	return num, nil
+}
```

Comment:
Some ideas on how we could improve this function:
* I'd suggest to not suppress the error returned by `strconv.Atoi`. We can join it via `errors.Join`.
* The `max(..., min..)` logic is quite obscure, we could simplify it
* We could improve the name of the function to make it clear it will parse and check that the number is `[1, maxValue]`

```suggestion
func parseNonZeroPositiveNumber(value string, maxValue int) (int, error) {
	num, err := strconv.Atoi(value)
	if err != nil {
		return -1, errors.Join(errInvalidNumericValue, err)
	}
	if num < 0 || num > maxValue {
		return -1, errInvalidNumericValue
	}
	return num, nil
}
```